hubspot/tasks.py

```diff
@@ -68,14 +77,57 @@ def check_hubspot_api_errors():
         defaults={"checked_on": now_in_utc()}
     )
     last_timestamp = hubspot_timestamp(last_check.checked_on)
+
+    order_sync_cache = {}
+
     for error in get_sync_errors():
-        if error.get("errorTimestamp") > last_timestamp:
-            msg = "Hubspot error for {obj_type} id {obj_id}: {details}".format(
-                obj_type=error.get("objectType", "N/A"),
-                obj_id=error.get("integratorObjectId", "N/A"),
-                details=error.get("details", ""),
+        error_timestamp = error.get("errorTimestamp")
+        if error_timestamp > last_timestamp:
+            obj_type = (error.get("objectType", "N/A"),)
+            obj_id = parse_hubspot_id(error.get("integratorObjectId", "")) or "N/A"
+            error_type = error.get("type", "N/A")
+            details = error.get("details", "")
+            # Handle line syncs that may have failed because of pending order syncs
+            if "LINE_ITEM" in obj_type and error_type == "INVALID_ASSOCIATION_PROPERTY":
+                order_id = parse_hubspot_id(ASSOCIATED_DEAL_RE.search(details).group(1))
+                try:
+                    line = Line.objects.get(id=obj_id)
+                except ObjectDoesNotExist:
+                    log.error(
+                        "Missing LINE %s for hubspot error %s: %s",
+                        obj_id,
+                        error_type,
+                        details,
+                    )
+                    continue
+
+                if str(order_id) not in order_sync_cache:
+                    try:
+                        # Get sync data and cache it to check against potential other line errors
+                        sync_data = get_sync_status("DEAL", line.order.id)
+                        order_sync_cache = sync_data["lastProcessedAt"]
```

Comment:
The cache might not be necessary either, since there should only be one line per order anyway (though it's possible this might change at some point in the future).